### ISSUE TITLE
GitHub oAuth doesn't work due to null e-mail address unless you set a public e-mail address on your profile

### ISSUE THREAD

**Issue by jamie-34254** [CybercentreCanada/assemblyline-ui#293]
So I don't really think this is a bug but it might be worth documenting when using GitHub's oAuth you can't access the e-mail address on the user without setting it as your public e-mail address.

This breaks auto_properties checks such as:
```
                  auto_properties:
                      - field: email
                        pattern: ^.+?@kryptoslogic\.com$
                        type: role
                        value: admin
```

in a not very useful way:
```
expected string or bytes-like object
Traceback (most recent call last):
  File "/var/lib/assemblyline/.local/lib/python3.9/site-packages/assemblyline_ui/api/v4/authentication.py", line 519, in oauth_validate
    data = parse_profile(user_data, oauth_provider_config)
  File "/var/lib/assemblyline/.local/lib/python3.9/site-packages/assemblyline_ui/helper/oauth.py", line 98, in parse_profile
    if re.match(auto_prop.pattern, value):
  File "/usr/local/lib/python3.9/re.py", line 191, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or bytes-like object
```
since the email is null.

This Stackoverflow answer details a little bit about it: https://stackoverflow.com/a/35387123 but basically the email field on that endpoint is only available if you set a public email address. Not really sure on the best way for you to handle this on your end, catching the TypeError on regex would be useful though.

Some documentation to cover it might be useful too, this config works for GitHub Enterprise, changing out the API endpoint for `api.github.com` on base and user_get and the auth/access to github.com would allow oAuth from GitHub.com although untested.

```
              github:
                  auto_create: true
                  auto_sync: true

                  client_id: idhere
                  client_secret: secrethere

                  access_token_url: https://ghe_hostname/login/oauth/access_token
                  api_base_url: https://ghe_hostname/
                  authorize_url: https://ghe_hostname/login/oauth/authorize

                  client_kwargs:
                      scope: read:user

                  user_get: https://ghe_hostname/api/v3/user

                  auto_properties:
                      - field: email
                        pattern: ^.+?@kryptoslogic\.com$
                        type: role
                        value: user

```

**Comment by cccs-sgaron** [comment]
I think we can definitely classify this as a bug. None type should be caught before doing the regex test. Will fix this today.

**Comment by cccs-sgaron** [comment]
This will be fix by PR #294. 

Basically, if the field mentioned by the auto_property is not in the profile or the field is the profile is None, auto_properties checks will be skipped for that field.
